charts/team-ns/templates/argocd/argocd-applicationset.yaml

```diff
@@ -0,0 +1,50 @@
+{{- $v := .Values }}
+{{- if $v.apps.argocd.enabled }}
+{{- range $v.workloads }}
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: team-{{ $v.teamId }}-{{ .name }}
+  namespace: argocd # ApplicationSets must always be in the argocd namespace
+spec:
+  generators:
+  - git:
+      repoURL: https://gitea.{{ $v.cluster.domainSuffix }}/otomi/values.git
+      revision: HEAD
+      files:
+      - path: "env/teams/workloads/{{ $v.teamId }}/{{ .name }}.yaml"
+  template:
+    metadata:
+      name: 'team-{{ $v.teamId }}-{{ .name }}'
+    spec:
+      project: team-{{ $v.teamId }} # project is restricted
+      source:
+        repoURL: '{{ .url }}'
+        targetRevision: '{{ .revision }}' # if using Helm, must be the revision in Chart.yaml
+        {{ with .path }}
+        path: '{{ . }}' # path to the helm chart git-hosted Helm charts
+        {{- end }}
+        {{ with .chart }}
+        chart: '{{ . }}'  # Set this only when pulling directly from a Helm repo. DO NOT set for git-hosted Helm charts.
+        {{- end }}
+        helm:
+          # The argocd template the values parameter is taken from the git generator path
+          values: '{{ "{{values}}" }}'
+          releaseName: '{{ .name }}'
+      syncPolicy:
+        automated:
+          allowEmpty: false
+          prune: true
+          selfHeal: true
+        retry:
+          backoff:
+            duration: 5s
+            factor: 2
+          limit: 100
```

Comment:
Thanks for feedback.
Interesting, Is the factor parameter a power of multiple?

I don't really understand how prune with combination of self heal could remove a PV. Good helm charts never removes the PVC. Could you elaborate?

Reply:
Yep factor is indeed exponential - from the docs: "Factor multiplies the base duration after each failed retry (default 2)". The current setting would allow for 100 sync attempts of 25 seconds each, which is too many attempts with each being too short. 

Regarding prune - I've done some research on this and found that with prune enabled then only if a resource is no longer defined in Git then it will be deleted, and not if it simply fails to sync - so mostly good there, except for the case where you are migrating from one storage type to another i.e. disk to S3 - much of the time you wouldn't want the old disk to be deleted (e.g. if it had log data).

What I would recommend is adding the sync params to the schema and letting the user specify their own settings, and also the option to disable sync altogether and do manual sync only - we don't want to box the user into sync settings that may not be suitable for their workload.

Reply:
ok

Reply:
thanks. parameters updated